eip_operator/src/controller/node.rs

```diff
@@ -47,8 +50,18 @@ impl k8s_controller::Context for Context {
             self.namespace.as_deref().unwrap_or("default"),
         );
 
+        // Check for unresponsive egress nodes and cleanup their egress status.
+        // This check must be done before any checks for EIPS. The EIPS may be
+        // already unassociated from the node.
+        let node_api = Api::<Node>::all(client.clone());
+        cleanup_old_egress_nodes(&eip_api, &node_api).await?;
```

Comment:
putting this here feels like it's going against the standard controller grain, and IMO it makes it much more complex how to reason about the flow.

ideally this `apply` function can reason locally about the health and EIP status of this Node and nothing more -- if it's healthy and matches an EIP resource, take on the assignment. If it's unhealthy and has an EIP, disassociate it.

by introducing `cleanup_old_egress_nodes` here we're now reasoning about the state of all nodes each time we reconcile any one node, which means having to hold a lot more in your head and introduces new race conditions

if it's possible, the more I'm trying to think through the scenarios here, the more I think we should consider ways to separate out the egress gateway label from this Eip/Node logic

Reply:
could we pull this logic out into its own control loop? we could watch for `Eip` resources. each time we see a status change to the `Eip`, we fetch the set of Nodes it applies to and update the gateway status label on each

this also more cleanly separates out EIP handling from egress gateway specific work

Reply:
I moved `cleanup_old_egress_nodes()` into it's own task in [this commit](https://github.com/MaterializeInc/k8s-eip-operator/pull/474/commits/c0dd1035983713762379599343e5f41173426e38). Next I'm working on:
* moving the `add_gateway_status_label()` ready labeling out of the eip node controller
* moving associated unused functions from the eip node controller file to a new file